scm/src/main/java/com/sandboni/core/scm/cli/ProcessRunner.java

```diff
@@ -0,0 +1,62 @@
+
+package com.sandboni.core.scm.cli;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class ProcessRunner {
+
+    private static final int DEFAULT_TIMEOUT_VALUE = 5;
+    private static final TimeUnit DEFAULT_TIMEOUT_UNITS = TimeUnit.SECONDS;
+
+    private static Process start(ProcessBuilder pb) throws IOException {
+        try {
+            return pb.start();
+        } catch (IOException e) {
+            throw new IOException("Unable to start sub-process", e);
+        }
+    }
+
+    private static void waitFor(Process p, long timeout, TimeUnit timeoutUnit) throws IOException {
+        try {
+            boolean finished = p.waitFor(timeout, timeoutUnit);
+            assert finished;
+        } catch (InterruptedException e) {
+            throw new IOException("Error running process", e);
+        }
+    }
+
+    public static List<String> runCommand(File workingDirectory, String... command) throws IOException {
+        return runCommand(workingDirectory, DEFAULT_TIMEOUT_VALUE, DEFAULT_TIMEOUT_UNITS, command);
+    }
+
+    private static List<String> runCommand(File workingDirectory, long timeout, TimeUnit timeoutUnit, String... command) throws IOException {
+        ProcessBuilder pb = new ProcessBuilder(command);
+
+        if (workingDirectory != null) {
+            pb.directory(workingDirectory);
+        }
+
+        pb.redirectErrorStream(true);
```

Comment:
Keep consistent line formatting for all your code

Reply:
ok